ape_infura/providers.py

```diff
@@ -18,32 +35,38 @@ class InfuraProviderError(ProviderError):
 
 
 class MissingProjectKeyError(InfuraProviderError):
-    def __init__(self):
-        env_var_str = ", ".join([f"${n}" for n in _ENVIRONMENT_VARIABLE_NAMES])
+    def __init__(self, options: Tuple[str, str]):
+        env_var_str = ", ".join([f"${n}" for n in options])
         super().__init__(f"Must set one of {env_var_str}")
 
 
 class Infura(Web3Provider, UpstreamProvider):
-    network_uris: Dict[str, str] = {}
+    network_uris: Dict[Tuple[str, str], str] = {}
 
     @property
     def uri(self) -> str:
+        ecosystem_name = self.network.ecosystem.name
         network_name = self.network.name
-        if network_name in self.network_uris:
-            return self.network_uris[network_name]
+        if (ecosystem_name, network_name) in self.network_uris:
+            return self.network_uris[(ecosystem_name, network_name)]
 
         key = None
-        for env_var_name in _ENVIRONMENT_VARIABLE_NAMES:
+        options = _ENVIRONMENT_VARIABLE_OPTIONS[ecosystem_name]
+        for env_var_name in options:
             env_var = os.environ.get(env_var_name)
             if env_var:
                 key = env_var
                 break
 
         if not key:
-            raise MissingProjectKeyError()
-
-        network_uri = f"https://{self.network.name}.infura.io/v3/{key}"
-        self.network_uris[network_name] = network_uri
+            raise MissingProjectKeyError(options)
+
+        prefix = f"{ecosystem_name}-" if ecosystem_name != "ethereum" else ""
+        network = self.network.name
+        if ecosystem_name == 'arbitrum' and self.network.name == 'testnet':
+            network = 'rinkeby'  # currently only "testnet is supported on arbitrum"
+        network_uri = f"https://{prefix}{self.network.name}.infura.io/v3/{key}"
```

Comment:
```suggestion
        network = self.network.name
        if ecosystem_name == 'arbitrum' and network_name == 'testnet':
            network_name = 'rinkeby'  # currently only "testnet is supported on arbitrum"
        network_uri = f"https://{prefix}{network_name}.infura.io/v3/{key}"
```

Reply:
nice catch, forgot about that variable, thank you! had to do it slightly the other way though, wrote in code comment why